README.md

```diff
@@ -344,13 +369,18 @@ activity that was originally called by ``startActivityForResult()``,
 the IDP app cannot change the activity before redirecting back to
 the RP app. That is a problem in cases where the IDP app uses different
 activities to log the user in or to request a 2-factor authentication.
+
+JJH: I don't follow this - if the iDP app wants to use extra activities, it can just retrieve the packageName in the first one and pass the string to the second activity?
```

Comment:
No this is not possible because the method ``setResult(int resultCode, Intent data)`` takes the packageName from the current context and as far as I know it is not possible to manually inject the packageName into the context. 

So if you use this method after the app navigated to another activity in the IDP app, you will jump back to the previous activity in the IDP app instead of jumping back to the RP app. And if the IDP app removes its own activities from the call stack, the ``setResult`` method will navigate the user back to the RP app, but for the RP app it looks like the request is canceled and there are no data.

Reply:
Ahhh. I'd misunderstood the text.

I did some research - looking at https://stackoverflow.com/questions/10407159/how-to-manage-startactivityforresult-on-android it looks like it is possible to launch further activities?

Reply:
As far as I understood the discussion on Stackoverflow they only discussed how to return different results back to an activity and how to launch different activities from one activity with ``startActivityForResult()``. But I talked about the following flow:  **RP_Activity** -- startActivityForResult() --> **IDP_Activity_0** -- finish(); startActivity() --> **IDP_Activity_1** -- finish(); startActivity() --> **IDP_Activity_0** -- setResult() --> **RP_Activity**. This flow doesn't work because when we go from IDP_Activity_1 to IDP_Activity_0 the IDP_Activity_0 has lost its state from before. If this is important I can build a minimal working example that shows this behavior. I don't say that developers can't find workarounds for this because it would be possible to use a Fragments instead of an Activity for IDP_Activity_1 but this would complicate the integration into existing apps I think.

Reply:
My understanding is IDP_Activity_0 must not finish until it returns the result. It can launch sub activities to handle it things - i.e. where you have 'finish' and 'start' I'd expected 'pause' and 'resume':

https://developer.android.com/reference/android/app/Activity#activity-lifecycle

There may still be some existing idp apps that find it difficult to provide a `startActivityForResult()`, but I wouldn't expect it to be that common, my understanding is Android programmers in general understand how to do it.

Reply:
It could work if we don't finish the Activity. But is it really worth the increased complexity? This repo also contains the document [AppAuth-Integration](https://github.com/oauthstuff/app2app-evolution/blob/master/AppAuth-Integration.md) where I describe how to integrate the app2app redirection into the AppAuth-Android project. Without ``startActivityForResult()`` we can provide one method to securely redirect from the RP to the IDP and from the IDP back to the RP.

Reply:
Any final thoughts on this before we merge? I think it is especially important for the API I would integrate into the AppAuth-Android project to decide whether we want to use ``startActivityForResult()`` or not. This API is explained in this [document](https://github.com/oauthstuff/app2app-evolution/blob/master/AppAuth-Integration.md). I think it would be better not to use ``startActivityForResult()`` to simplify the implementation of app2app.